app/scripts/shell/dimAppCtrl.controller.js

```diff
@@ -162,27 +162,35 @@
       }
     };
 
+    // Don't refresh more than once a minute
+    var refresh = _.throttle(vm.refresh, 60 * 1000);
+
     vm.startAutoRefreshTimer = function () {
       var secondsToWait = 360;
 
       $rootScope.autoRefreshTimer = $interval(function () {
        //Only Refresh If We're Not Already Doing Something
        //And We're Not Inactive
-       if (!$rootScope.loadingTracker.active() && !$rootScope.isUserInactive()) {
-         vm.refresh();
+       if (!$rootScope.loadingTracker.active() && !$rootScope.isUserInactive() && document.visibilityState == 'visible') {
+         refresh();
        }
       }, secondsToWait * 1000);
     };
 
     vm.startAutoRefreshTimer();
 
-    var refresh = _.debounce(vm.refresh, 300);
-
     $rootScope.$on('dim-settings-updated', function(event, arg) {
       // if ((!_.has(arg, 'charCol')) && (!_.has(arg, 'vaultCol'))) {
         refresh();
       // }
     });
+
+    // Refresh when the user comes back to the page
+    document.addEventListener("visibilitychange", function() {
```

Comment:
Hm, it's been plenty fast for me, but I have a fast computer. I think the recent reductions in watches and my change to reuse item ids helped a lot too. 

-Ben

Sent from my phone

> On Oct 20, 2015, at 9:50 AM, kyle shay <notifications@github.com> wrote:
> 
> In app/scripts/shell/dimAppCtrl.controller.js:
> 
> >      $rootScope.$on('dim-settings-updated', function(event, arg) {
> >        // if ((!_.has(arg, 'charCol')) && (!_.has(arg, 'vaultCol'))) {
> >          refresh();
> >        // }
> >      });
> > +
> > +    // Refresh when the user comes back to the page
> > +    document.addEventListener("visibilitychange", function() {
> I love the idea of refreshing when a user comes back. But, I don't want to force it. We're getting some complaints of 'slowness' in DIM. Refreshing the items takes a bit of time, and adds to the sluggishness.
> 
> I talk a bit about it in #203, Ideally I would want a toaster to appear and if the user clicks that toaster it could refresh (or something like that.) Otherwise, it should be a setting that a user can turn on.
> 
> ―
> Reply to this email directly or view it on GitHub.
>